cmd/rpcdaemon/commands/eth_receipts.go

```diff
@@ -328,6 +329,13 @@ func marshalReceipt(receipt *types.Receipt, txn types.Transaction) map[string]in
 		"logsBloom":         types.CreateBloom(types.Receipts{receipt}),
 	}
 
+	if !chainConfig.IsLondon(block.NumberU64()) {
+		fields["effectiveGasPrice"] = hexutil.Uint64(txn.GetPrice().Uint64())
+	} else {
+		baseFee, _ := uint256.FromBig(block.BaseFee())
+		gasPrice := new(big.Int).Add(block.BaseFee(), txn.GetEffectiveGasTip(baseFee).ToBig())
+		fields["effectiveGasPrice"] = hexutil.Uint64(gasPrice.Uint64())
```

Comment:
Do you think need create hexutil.Uint256() method?

Reply:
The main reason for using hexutil.Uint64 is because I think EffectiveGasPrice type might be uint64, 
but I see the GasPrice uses uint256. 
Is there any specific reason for using uint256? 
And how to do the "overflow" check here?
I think it would be nice to have hexutil.Uint256().

Before:
```
baseFee, _ := uint256.FromBig(block.BaseFee())
gasPrice := baseFee.Add(baseFee, txn.GetEffectiveGasTip(baseFee))
fields["effectiveGasPrice"] = hexutil.Uint64(gasPrice.Uint64())
```

After:
```
baseFee, _ := uint256.FromBig(block.BaseFee())
gasPrice := baseFee.Add(baseFee, txn.GetEffectiveGasTip(baseFee))
fields["effectiveGasPrice"] = hexutil.Uint256(gasPrice)
```

Reply:
`Is there any specific reason for using uint256?` - we just using uint256 everywhere where go-ethereum using big.Int (because it's more compatible with ETH spec/paper). But why GasPrice is big.Int (answer is as usually "security"): 1ETH it's 2^60 Wei - which is already almost MaxUint64=2^64. It means using Uint64 for any operations with Wei is bad idea - in general - but in this case it's fine.

How to handle overflow - just return error on overflow.